### DISCUSSION TITLE
Item metadata management by Fabric

### DISCUSSION THREAD

**Discussion by arajawat** [microsoft/Microsoft-Fabric-workload-development-sample#53] [Help]
When a workload item is created in local devbox (using gateway),  the boilerplate backend service is writing the item metadata to my local folder. However, when I stop the backend service and navigate to the workspace, I can still see the custom items created. Does that mean that information flows in following manner leads to creation of item metadata in Fabrics metadata database as well?

Workload FE (Create Item) -> Fabric FE -> Fabric BE (Any storage here?) -> Workload BE (Store Item metadata)

Few other related questions:
1. Can we assume that any CRUD operation on workload item leads to item metadata CRUD in Fabric metadata database as well?
2. If yes, How is the item metadata managed within Fabric, can we access it progmatically?
3. Lets say a user deletes the workload item directly from workspace, we would want to clean up the relevant item metadata from Workload Backend as well. How can we do that? Since delete operation has been done from workspace and not from Workload FE IFrame. Is there a way that workload backend be notified about such operations that were not trigerred from workload FE?

**Comment by omgharra** [comment]
Hi @arajawat,

The Fabric backend does not store item metadata; it only forwards requests to your local workload backend. If a user deletes an item directly from the workspace, there is no built-in mechanism to notify your workload backend. You may need to implement a custom solution, like event listeners or a monitoring service, to handle such scenarios and clean up metadata accordingly.

**Comment by arajawat** [comment]
If that is the case, How am I able to see the Sample workload created in my workspace even after shuting down backend service and Dev gateway? I can see this with only FrontEnd service running:

![image](https://github.com/user-attachments/assets/51f6b00c-9c7d-4dec-8191-5d0bfc6cbce1)

**Reply by omgharra** [reply]
You'll still see it in your workspace even if dev gateway or BE service are down. They won't be clickable and Play icon would disappear.

**Comment by omgharra** [comment]
Discussed offline - I believe this can be closed

**Comment by arajawat** [comment]
Sure, Thanks.
